packages/react-sdk/src/client.ts

```diff
@@ -0,0 +1,595 @@
+      logger.info(
+        'getVariation returned null for experiment "%s" because userId is not set',
+        experimentKey,
+      )
+      return null
+    }
+    return this.client.getVariation(experimentKey, userId, userAttributes)
+  }
+
+  track(
+    eventKey: string,
+    overrideUserId?: string | EventTags,
+    overrideAttributes?: UserAttributes,
+    eventTags?: EventTags,
+  ) {
+    if (typeof overrideUserId !== 'undefined' && typeof overrideUserId !== 'string') {
+      eventTags = overrideUserId
+      overrideUserId = undefined
+      overrideAttributes = undefined
+    }
+    const [userId, attributes] = this.getUserIdAndAttributes(
+      overrideUserId,
+      overrideAttributes,
+    )
+
+    if (userId === null) {
+      logger.info(
+        'track for event "%s" not being sent because userId is not set',
+        eventKey,
+      )
+      // TODO log
+      return
+    }
+
+    return this.client.track(eventKey, userId, attributes, eventTags)
+  }
+
+  /**
+   * Note: in the case where the feature isnt in the datafile or the datafile hasnt been
+   * loaded, this will return `false`
+   *
+   * @param {string} feature
+   * @param {string} userId
+   * @param {UserAttributes} [attributes]
+   * @returns {boolean}
+   * @memberof OptimizelySDKWrapper
+   */
+  public isFeatureEnabled(
+    feature: string,
+    overrideUserId?: string,
+    overrideAttributes?: UserAttributes,
+  ): boolean {
+    const [userId, userAttributes] = this.getUserIdAndAttributes(
+      overrideUserId,
+      overrideAttributes,
+    )
+    if (userId === null) {
+      logger.info(
+        'isFeatureEnabled returning false for feature "%s" because userId is not set',
+        feature,
+      )
+      return false
+    }
+    return this.client.isFeatureEnabled(feature, userId, userAttributes)
+  }
+
+  /**
+   * Get all variables for a feature, regardless of the feature being enabled/disabled
+   *
+   * @param {string} feature
+   * @param {string} userId
+   * @param {UserAttributes} [attributes]
+   * @returns {VariableValuesObject}
+   * @memberof OptimizelySDKWrapper
+   */
+  public getFeatureVariables(
+    featureKey: string,
+    overrideUserId?: string,
+    overrideAttributes?: UserAttributes,
+  ): VariableValuesObject {
+    const [userId, userAttributes] = this.getUserIdAndAttributes(
+      overrideUserId,
+      overrideAttributes,
+    )
+    if (userId === null) {
+      logger.info(
+        'getFeatureVariables returning `{}` for feature "%s" because userId is not set',
+        featureKey,
+      )
+      return {}
+    }
+    let variableObj: { [key: string]: any } = {}
+    const config = (this.client as any).projectConfigManager.getConfig()
+    if (!config) {
+      return {}
+    }
+    const feature = config.featureKeyMap[featureKey]
+    if (!feature) {
+      return {}
+    }
+    let variables: object[] = feature.variables
+    variables.forEach((variableDef: any) => {
+      let type: any = variableDef.type
```

Comment:
Aren't both of these `string`s? Any reason why we are treating them as `any`?

Reply:
They are strings in reality, but not part of the public interface, and thus not part of the type definition for the optimizely `Client`. That's why we have to cast to `any` on line 356. We explicitly declare these variables as having type `any` as a way of calling them out. If variables are implicitly `any` they can leak all over the place and seriously harm type safety.